core/src/main/scala/akka/persistence/r2dbc/internal/sqlserver/SqlServerDialectHelper.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal.sqlserver
+
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.internal.InstantFactory
+import com.typesafe.config.Config
+import io.r2dbc.spi.Row
+
+import java.time.Instant
+import java.time.LocalDateTime
+import java.util.TimeZone
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[r2dbc] object SqlServerDialectHelper {
+  def apply(config: Config) = new SqlServerDialectHelper(config)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[r2dbc] class SqlServerDialectHelper(config: Config) {
+
+  private val tagSeparator = config.getString("tag-separator")
+
+  require(tagSeparator.length == 1, s"Tag separator '$tagSeparator' must be a single character.")
+
+  def tagsToDb(tags: Set[String]): String = {
+    if (tags.exists(_.contains(tagSeparator))) {
```

Comment:
slightly more efficient to use Char as tagSeparator, since we only allow single character